### ISSUE TITLE
Game crash when sprinting cat touches water

### ISSUE THREAD

**Issue by mc-schmitty** [Ladysnake/Requiem#360]
Hi, haven't submitted one of these before so I hope this helps.

Minecraft gets a NoSuchElementException when a cat sprints (scared from a player) into water or an already floating cat attempts to sprint in water. 

Here's the crash log:
```
[22/06/2021 16:11:16 PM] [Server thread/ERROR]: Encountered an unexpected exception
[22/06/2021 16:11:16 PM] net.minecraft.class_148: Ticking entity
[22/06/2021 16:11:16 PM] 	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:875) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:811) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1132.method_3748(class_1132.java:91) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:670) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at java.lang.Thread.run(Unknown Source) [?:1.8.0_251]
[22/06/2021 16:11:16 PM] Caused by: java.util.NoSuchElementException: class_1451['Cat'/76, l='ServerLevel[AAAAAAAA]', x=572.53, y=62.40, z=170.68] provides no component of type requiem:movement_alterer
[22/06/2021 16:11:16 PM] 	at nerdhub.cardinal.components.api.ComponentType.get(ComponentType.java:109) ~[a262ff81-9323-4a87-809f-979841c4cbcc.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1309.handler$zgl000$preventWaterHovering(class_1309.java:5977) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1309.method_26317(class_1309.java:2155) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1309.method_6091(class_1309.java:1995) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1309.method_6007(class_1309.java:2547) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1308.method_6007(class_1308.java:542) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1296.method_6007(class_1296.java:127) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1429.method_6007(class_1429.java:54) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1309.method_5773(class_1309.java:2254) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1308.method_5773(class_1308.java:342) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1451.method_5773(class_1451.java:303) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_3218.method_18762(class_3218.java:621) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_1937.method_18472(class_1937.java:561) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.class_3218.method_18765(class_3218.java:411) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:871) ~[intermediary-client-1.16.5.jar:?]
[22/06/2021 16:11:16 PM] 	... 5 more
```

Here's the link to the complete ATLauncher log with more details: https://paste.atlauncher.com/view/0f702d7d-b136-45f7-9057-552b8b3fd00c
[22/06/2021 15:39:40 PM] is when I first encounter the crash, with all mods enabled,
[22/06/2021 16:11:16 PM] is when I reproduce the crash with all mods disabled except for the Fabric api and Requiem.

It seems like someone else may have also encountered this crash, but their problem might be different: https://www.reddit.com/r/Minecraft/comments/nzrld5/help_is_this_saying_my_pet_cat_is_crashing_the/
(Also their real life cat peed on their keyboard, which isn't something I needed to reproduce this crash)

I hope I included everything! Let me know if the links aren't working or anything.

**Comment by Pyrofab** [comment]
Thanks a bunch for the report ! I had never heard of it before, but I can see why it would happen. That said, we are currently focusing on 1.17 development, so I can't promise a fix on 1.16.

**Comment by teidova** [comment]
Hello,

I'm currently using requiem on a 1.16.5 server and we love it but this error causes repetitive crashes so we are thinking of removing it. :(
People are complaining about rollbacks due to fatal crashes some lost important amount of stuff already :/

If you happen to remove it, please let us know :eyes: and we'll happily reinstall ! 👀

Thanks,
teidova21
